### DISCUSSION TITLE
can it work like a separate component and file

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#667] [Q&A]
Can you use system components like react js and vue js . To fill elements . And is the process and emit between parent component and child component 

Example in react js

Import HOME from home.jsx

<Parent component>
<Home/>

And use home  Inside the parent.js file

**Comment by rodja** [comment]
Same as @falkoschindler wrote in https://github.com/zauberzeug/nicegui/discussions/666#discussioncomment-5467523, I would like you to be more specific, provide full executable examples and explain your goal. What have you tried already? What is failing? Currently the issue is too cryptic and broad to help you along.

**Comment by bobwatcherx** [comment]
So like components on vue js . Where the code will be loaded from other file

Example
from nicegui import ui
from home import homepage

homepage()


ui.run()

Can it be like that or is there another way. To be similar to quasar js . And how to pass props and emit from child to parent component

**Reply by falkoschindler** [reply]
Sure. Every Python file is a "module". So you can modularize a NiceGUI app like every other Python program.
See https://github.com/zauberzeug/nicegui/discussions/661 for a similar discussion and additional references.
I'll close this thread as duplicate.
